c7n/resources/opsworks.py

```diff
@@ -57,6 +60,55 @@ class resource_type(object):
         date = 'CreatedAt'
         dimension = "StackId"
 
+    retry = staticmethod(get_retry(('ThrottlingException',)))
+    permissions = ('opsworks:ListTags',)
+
+    def augment(self, stacks):
+        filter(None, _opsworks_tags(
+            stacks, self.session_factory, self.executor_factory, self.retry
+        ))
+        return stacks
+
+def _opsworks_tags(stacks, session_factory, executor_factory, retry):
```

Comment:
this function should be encapsulated within the OpsworkStack resource itself

Reply:
fixed